src/Microsoft.AspNetCore.Routing/Patterns/DefaultRoutePatternTransformer.cs

```diff
@@ -53,16 +53,7 @@ private RoutePattern SubstituteRequiredValuesCore(RoutePattern original, RouteVa
                     // 1. Required value is null-ish - check to make sure that this route doesn't have a
                     // parameter or filter-like default.
 
-                    if (original.GetParameter(kvp.Key) != null)
```

Comment:
This test caused issues with action routes that contained {page} and page was a required value because of Razor pages.

Reply:
I'm not sure I get this. Does that work at all in MVC today? I don't understand why we would change a routing feature for this.

Reply:
You're right, this change is a bug. REVERTED